File: Examples/Lesson 6/House_ObjC_main/House_ObjC_main/House.m
//
//  House.m
//  HouseClass
//
//  Created by Gabrielle Miller-Messner on 2/23/16.
//  Copyright © 2016 Gabrielle Miller-Messner. All rights reserved.
//

#import "House.h"
//#import "House_ObjC_main-Swift.h"

@interface House()

@property (nonatomic, readwrite) int numberOfBedrooms;

@end

@implementation House

-(instancetype)initWithAddress: (NSMutableString*)address {
    self = [super init];
    
    if(self) {
        _address = [address copy];
        _numberOfBedrooms = 3;
        _hasHotTub = false;
    }
 
    return self;
}

-(instancetype)init {
    self = [super init];
    
    if(self) {
        _hasHotTub = true;
        _hotTub = [[HotTub alloc] init];
    }
    
    return self;
}


@end

File: Examples/Lesson 6/House_Swift_main/House_Swift_main/House.swift
//
//  House.swift
//  House_main_Swift
//
//  Created by Gabrielle Miller-Messner on 5/9/16.
//  Copyright © 2016 Gabrielle Miller-Messner. All rights reserved.
//

import Foundation

class House {
    var address = "555 Park Ave"
    let numberOfBedrooms = 3
    var hasHotTub = false
    var hotTub: HotTub?

    init(address: String) {
        self.address = address
        self.hasHotTub = false
    }

    init() {
        self.hasHotTub = true
        self.hotTub = HotTub()
    }

}

